web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -150,39 +152,92 @@ function getNavSubsectionForRoute(
   };
 }
 
+/** useDebounce is used to add a slight delay to RightPanel transitions to prevent a section from collapsing if the user overshoots it or inadvertendly
+ * travels over another section, such as while moving their mouse to the RightPanel. For better UX, we skip the debounce delay when opening a new section
+ * from a closed state. */
+function useDebounce<T>(value: T, delay: number): T {
```

Comment:
we have a `debounce` in `highbar.ts`. Is that viable or do we need this new hook for some reason?

Reply:
looking through it, this hook I made it here is tailored to our use case and I don't think it's possible for the one in `highbar.ts` to do what we need to do